### ISSUE TITLE
Failed tests on clisp

### ISSUE THREAD

**Issue by catap** [3b/3bmd#61]
I'm using clisp from https://gitlab.com/gnu-clisp/clisp/-/commit/66924971790e4cbee3d58f36e530caa0ad568e5f and attempt to run tests via MacPorts leads to failure:
```
:info:test   INDENT-BY-TAB-SHOULD-BE-REPLACED-WITH-SPACES........................................................................[ OK ]
:info:test   BLANK-LINE-TEST2........................................................................[ OK ]
:info:test   BLANK-LINE-TEST1........................................................................[ OK ]
:info:test   NEWLINE........................................................................[ OK ]
:info:test   MULTIPLE-SPACES-MIXED-WITH-TABS........................................................................[ OK ]
:info:test   TAB-AS-SPACE........................................................................[ OK ]
:info:test   SPACE-TEST........................................................................[ OK ]
:info:test   EOF-TEST........................................................................[ OK ]
:info:test Test run had 1 failure:
:info:test   Failure 1: FAILED-ASSERTION when running 3BMD-TESTS::PARSE-LIST-WITH-CARRIAGE-RETURN
:info:test     Binary predicate (EQUALP X Y) failed.
:info:test     x: 3BMD-TESTS::RESULT => 
:info:test     ((:BULLET-LIST
:info:test       (:LIST-ITEM
:info:test        (:PLAIN "x"
:info:test         "
:info:test     "
:info:test         "y"
:info:test         "
:info:test     "
:info:test         "Not" " " "verbatim"))))
:info:test     y: 3BMD-TESTS::EXPECTED => 
:info:test     ((:BULLET-LIST
:info:test       (:LIST-ITEM
:info:test        (:PARAGRAPH
:info:test         "x
:info:test     y")
:info:test        (:PARAGRAPH "Not" " " "verbatim"))))
:info:test *** - tests failed
```

**Comment by 3b** [comment]
Hmm, I'm guessing it is at least partially Mac specific, and either git or clisp translated the carriage return in the test in some way that confused the test.

Not sure if it indicates any actual problem or if the test just needs to be specified differently to avoid translation.

@melisgl any ideas?

**Comment by melisgl** [comment]
Ah, yes. I the test is incorrect on mac and probably windows too. On mac, the correct version should be using newline instead of carriage return. I'll make a merge request in a couple of days for testing as I have no access to mac. In the meantime, can you perhaps verify that the same failure happens with SBCL, too?

**Comment by catap** [comment]
@melisgl tests are passing on the same macOS on`ECL 21.2.1` and `SBCL 2.3.7`. And I'm using CLisp from it's master.

**Comment by melisgl** [comment]
Got access to a mac in the meantime, but I'm having some trouble installing/compiling CLISP (no root access).
Could you please evaluate the expression below on CLISP and SBCL on the affected system and post the outputs here?

```
(let ((string (coerce '(#\x #\Return #\y) 'string)))
  (with-open-file (out "/tmp/pax-test" :direction :output
                                       :if-does-not-exist :create
                                       :if-exists :supersede)
    (write-string string out))
  (let ((reread (with-open-file (in "/tmp/pax-test")
                  (read-line in))))
    (loop for c across reread
          do (print (char-name c)))))
```


Also, what does locale say in the shell and what is the value of CUSTOM:*DEFAULT-FILE-ENCODING* in CLISP?

**Comment by catap** [comment]
Both just `clisp` and `env LC_ALL=C` returns:
```
"LATIN_SMALL_LETTER_X" 
NIL
```

**Comment by melisgl** [comment]
Argh. Try this then.

```
(ql:quickload "alexandria")

(let ((string (coerce '(#\x #\Return #\y) 'string)))
  (with-open-file (out "/tmp/pax-test" :direction :output
                                       :if-does-not-exist :create
                                       :if-exists :supersede)
    (write-string string out))
  (let ((reread (with-open-file (in "/tmp/pax-test")
                  (alexandria:read-file-into-string in))))
    (loop for c across reread
          do (print (char-name c)))))
```

**Comment by catap** [comment]
@melisgl I haven't use `quiclisp`, instead everything is installed via MacPorts. So, I've replaced it by:
```
(require :asdf)
(asdf:operate 'asdf:load-op 'alexandria)
```

And both way to call `clisp` returns:
```
"LATIN_SMALL_LETTER_X" 
"Newline" 
"LATIN_SMALL_LETTER_Y" 
NIL
```

**Comment by melisgl** [comment]
OK, now that I read up on history (https://en.wikipedia.org/wiki/Newline), I realize that the test is fine on modern macOS because only classic mac OS used to use carriage return for ending a line. I'm tempted to just to disable this test for CLISP (and for windows, where it is indeed broken).

**Comment by melisgl** [comment]
Turns out this happens on Linux+CLISP as well.

**Comment by melisgl** [comment]
Ahem. On CLISP:

```
[1]> (char-code #\newline)
10
[2]> (char-code #\linefeed)
10
[3]> (char-code #\return)
13
```

I'm pretty sure linefeed should be the same as return.

**Comment by melisgl** [comment]
Sorry about the confusion and the deleted messages. Here is a pull request: https://github.com/3b/3bmd/pull/62.

**Comment by 3b** [comment]
merged #62
